config/initializers/new_framework_defaults_7_1.rb

```diff
@@ -1,214 +0,0 @@
-# as equal to an equivalent `Hash` by default.
-Rails.application.config.action_controller.allow_deprecated_parameters_hash_equality = false
-
-# Active Record Encryption now uses SHA-256 as its hash digest algorithm. Important: If you have
-# data encrypted with previous Rails versions, there are two scenarios to consider:
-#
-# 1. If you have +config.active_support.key_generator_hash_digest_class+ configured as SHA1 (the default
-# before Rails 7.0), you need to configure SHA-1 for Active Record Encryption too:
-# Rails.application.config.active_record.encryption.hash_digest_class = OpenSSL::Digest::SHA1
-# 2. If you have +config.active_support.key_generator_hash_digest_class+ configured as SHA256 (the new default
-# in 7.0), then you need to configure SHA-256 for Active Record Encryption:
-# Rails.application.config.active_record.encryption.hash_digest_class = OpenSSL::Digest::SHA256
-#
-# If you don't currently have data encrypted with Active Record encryption, you can disable this setting to
-# configure the default behavior starting 7.1+:
-# Rails.application.config.active_record.encryption.support_sha1_for_non_deterministic_encryption = false
-
-# No longer run after_commit callbacks on the first of multiple Active Record
-# instances to save changes to the same database row within a transaction.
-# Instead, run these callbacks on the instance most likely to have internal
-# state which matches what was committed to the database, typically the last
-# instance to save.
-Rails.application.config.active_record.run_commit_callbacks_on_first_saved_instances_in_transaction = false
-
-# Configures SQLite with a strict strings mode, which disables double-quoted string literals.
-#
-# SQLite has some quirks around double-quoted string literals.
-# It first tries to consider double-quoted strings as identifier names, but if they don't exist
-# it then considers them as string literals. Because of this, typos can silently go unnoticed.
-# For example, it is possible to create an index for a non existing column.
-# See https://www.sqlite.org/quirks.html#double_quoted_string_literals_are_accepted for more details.
-Rails.application.config.active_record.sqlite3_adapter_strict_strings_by_default = true
-
-# Disable deprecated singular associations names
-Rails.application.config.active_record.allow_deprecated_singular_associations_name = false
-
-# Enable the Active Job `BigDecimal` argument serializer, which guarantees
-# roundtripping. Without this serializer, some queue adapters may serialize
-# `BigDecimal` arguments as simple (non-roundtrippable) strings.
-#
-# When deploying an application with multiple replicas, old (pre-Rails 7.1)
-# replicas will not be able to deserialize `BigDecimal` arguments from this
-# serializer. Therefore, this setting should only be enabled after all replicas
-# have been successfully upgraded to Rails 7.1.
-# Rails.application.config.active_job.use_big_decimal_serializer = true
```

Comment:
This one needs consideration as to what the impact would be of enabling now -vs- waiting for a proper 4.3 release running Rails 7.1 first (and then enable this in a future release).